src/server/transaction.cc

```diff
@@ -1091,7 +1061,7 @@ Transaction::RunnableResult Transaction::RunQuickie(EngineShard* shard) {
 void Transaction::ExpireBlocking(WaitKeysProvider wcb) {
   DCHECK(!IsGlobal());
   DVLOG(1) << "ExpireBlocking " << DebugId();
-  run_barrier_.Start(unique_shard_cnt_);
+  run_barrier_.Add(unique_shard_cnt_);
```

Comment:
in your embedded bc please add `Set` or `Start` function that actually dchecks that the previous value is 0.